libraries/FreeRTOS/src/variantHooks.cpp

```diff
@@ -70,8 +70,15 @@ extern "C" {
         xSemaphoreGive(mtx);
     }
 
-    void __freertos_mutex_give_from_isr(SemaphoreHandle_t mtx) {
-        xSemaphoreGiveFromISR(mtx, NULL);
+    void __freertos_mutex_give_from_isr(SemaphoreHandle_t mtx, bool portYield) {
+        if (!portYield) {
```

Comment:
Here, I'm a little fuzzy.  When we take the mutex, FreeRTOS returns a "higherTaskWoken" bool which,m if true, means we need to do a context switch on the giveFromISR.  That's the `portYield` param here.  https://www.freertos.org/xSemaphoreTakeFromISR.html

But the giveFromISR also seems to return the same flag:  https://www.freertos.org/xSemaphoreTakeFromISR.html  .

So, maybe the logic for this function should be:
````
BaseType_t pxHigherPriorityTaskWoken;
xSemaphoreGiveFromISR(mtx, &pxHigherPriorityTaskWoken);
portYIELD_FROM_ISR( (pxHigherPriorityTaskWoken || portYield) ? pdTRUE : pdFALSE);
````
?

Reply:
The pxHigherPriorityTaskWoken can be used multiple times.

From the docs

> FreeRTOS API functions can only set *pxHighPriorityTaskWoken to pdTRUE. If an ISR
calls more than one FreeRTOS API function, then the same variable can be passed as the
pxHigherPriorityTaskWoken parameter in each API function call, and the variable only
needs to be initialized to pdFALSE before it is used for the first time.